### PR TITLE
[vividus] Move status code calculation into results provider

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus/src/main/java/org/vividus/results/ResultsProvider.java [hunk 1]
```diff
@@ -20,14 +20,13 @@
...
 
-import org.vividus.report.allure.model.Status;
 import org.vividus.results.model.Failure;
 import org.vividus.results.model.NodeType;
 import org.vividus.results.model.Statistic;
 
 public interface ResultsProvider
 {
-    Optional<Status> getRunStatus();
+    int getStatusCode();
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/results/ResultsProvider.java:29]:
```suggestion
    int calculateExitCode();
```

File: vividus/src/main/java/org/vividus/runner/AbstractTestRunner.java [hunk 2]
```diff
@@ -32,14 +31,10 @@
...
 import org.vividus.log.TestInfoLogger;
-import org.vividus.report.allure.model.Status;
 import org.vividus.results.ResultsProvider;
 
 public abstract class AbstractTestRunner extends JUnitStories
 {
-    private static final int RUN_PASSED_EXIT_CODE = 0;
-    private static final int RUN_KNOWN_ISSUES_EXIT_CODE = 1;
-    private static final int RUN_FAILED_EXIT_CODE = 2;
     private static final int ERROR_EXIT_CODE = 3;
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/runner/AbstractTestRunner.java:38]: create an enum for all exit codes

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus/src/main/java/org/vividus/results/CollectingStatisticsStoryReporter.java [hunk 3]
```diff
@@ -104,9 +108,19 @@ public Optional<List<Failure>> getFailures()
...
+        if (runStatus.isEmpty())
+        {
+            return RUN_FAILED_EXIT_CODE;
+        }
+        switch (runStatus.get())
+        {
+            case PASSED:            return RUN_PASSED_EXIT_CODE;
+            case KNOWN_ISSUES_ONLY: return RUN_KNOWN_ISSUES_EXIT_CODE;
+            default:                return RUN_FAILED_EXIT_CODE;
+        }
```
Comment (ikalinin1) [inline vividus/src/main/java/org/vividus/results/CollectingStatisticsStoryReporter.java:123]:
```suggestion
       return status.get().map(s -> {
          switch (s)
          {
              case PASSED:            return RUN_PASSED_EXIT_CODE;
              case KNOWN_ISSUES_ONLY: return RUN_KNOWN_ISSUES_EXIT_CODE;
              default:                return RUN_FAILED_EXIT_CODE;
          }
        }).orElse(RUN_FAILED_EXIT_CODE)
     
```

File: vividus/src/main/java/org/vividus/results/ResultsProvider.java (hunk 1 above)
Comment (uarlouski) [reply vividus/src/main/java/org/vividus/results/ResultsProvider.java:29]: this is exit code in the context of System.exit(...) but as per docs it also called status code that less coupled to exit op

File: vividus/src/main/java/org/vividus/results/CollectingStatisticsStoryReporter.java (hunk 3 above)
Comment (uarlouski) [reply vividus/src/main/java/org/vividus/results/CollectingStatisticsStoryReporter.java:123]: prefer initial variant as the switch in map doesn't look good (just my opinion)

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**